### PR TITLE
GDB-10266 Set the `asg_enable_instance_refresh` to default to false

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description

Set the `asg_enable_instance_refresh` to default to false
Improved logging by adding a timestamp in front of each message. Updated README.md
Updated the README.md

## Related Issues

GDB-10266

## Changes

Set the `asg_enable_instance_refresh` to default to false
Improved logging by adding a timestamp in front of each message. Updated README.md
Updated the README.md
Added new template file 00_functions.sh containing helper functions
Added timeout mechanism when waiting for available volume during instance refresh in 01_wait_node_count.sh.tpl
Added new mechanism to check for unresponsive nodes defined in Route53 and update their IPs instead of creating new DNS record in 03_dns_provisioning.sh.tpl
Changed the way the node_count is obtained.
Added retry mechanism when joining the cluster if the initial join fails for any reason it will retry 2 more times with 5 min delay

## Screenshots (if applicable)

N/A

## Checklist

- [x] I have tested these changes thoroughly.
- [x] My code follows the project's coding style.
- [x] I have added appropriate comments to my code, especially in complex areas.
- [x] All new and existing tests passed locally.

### CODE REVIEW

File: modules/graphdb/templates/00_wait_node_count.sh.tpl [hunk 1]
```diff
@@ -11,6 +11,11 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# Function to print messages with timestamps
+log_with_timestamp() {
```
Comment (mihailradkov) [inline modules/graphdb/templates/00_wait_node_count.sh.tpl:15]: As a future improvement, you could add `log_info`, `log_error` etc.

Comment (viktor-ribchev) [reply modules/graphdb/templates/00_wait_node_count.sh.tpl:15]: Noted

Comment (mihailradkov) [inline modules/graphdb/templates/00_wait_node_count.sh.tpl:15]: Try defining this once for all scripts, e.g. `00_helpers.sh` and then use `source ...` to load it in the beginning.

Comment (viktor-ribchev) [reply modules/graphdb/templates/00_wait_node_count.sh.tpl]: done

**Pull request merged**